vllm/sampling_params.py

```diff
@@ -246,10 +253,10 @@ def clone(self) -> "SamplingParams":
         See https://github.com/vllm-project/vllm/issues/3087
         """
 
-        logit_processor_refs = None if self.logits_processors is None else {
+        logit_processor_refs = (None if self.logits_processors is None else {
             id(lp): lp
-            for lp in self.logits_processors
-        }
+            for lp in self.logits_processors if not hasattr(lp[0], "fsm")
```

Comment:
This affects the following [line](https://github.com/vllm-project/vllm/blob/abfc4f3387c436d46d6701e9ba916de8f9ed9329/vllm/engine/llm_engine.py#L271)  and as described in the following [issue](https://github.com/vllm-project/vllm/issues/3087) and partially addressed by the following [PR](https://github.com/vllm-project/vllm/pull/3099). Unfortunately, in this case, the copy of logits_processors is inevitable and seems like it indeed does slow down the inference. However, I'm using a relatively old GPU (4x NVIDIA RTX A4000, 16Gb) so I would need someone with better hardware to benchmark the speed. Refer to the [issue](https://github.com/vllm-project/vllm/issues/3448) for the code snippets to reproduce the tests